src/rosdep2/sources_list.py

```diff
@@ -558,8 +558,14 @@ def load_cached_sources_list(sources_cache_dir=None, verbose=False):
         if verbose:
             print('no cache index present, not loading cached sources', file=sys.stderr)
         return []
-    with open(cache_index, 'r') as f:
-        cache_data = f.read()
+    try:
+        with open(cache_index, 'r') as f:
+            cache_data = f.read()
+    except IOError as e:
+        if e.strerror == 'Permission denied':
+            raise CachePermissionError('Failed to write cache file: ' + str(e))
```

Comment:
It would be a good idea to preserve the original context here.
```suggestion
            raise CachePermissionError('Failed to write cache file: ' + str(e)) from e
```

Reply:
This appears to cause syntax errors in python2. rosdep doesn't currently depend on six or any other such wrappers that would bridge this gap for us.

It seems like `exec("raise...")` could do what we want for only python3. I'm not afraid of it but it'll be the first time that exec/eval are used in rosdep outside of some tests.

Reply:
I completely forgot that this wasn't valid in Python 2. Just forget about it - it was just a Nice To Have.